mojo/stdlib/test/builtin/test_value.mojo

```diff
@@ -128,6 +153,113 @@ def test_type_inherit_non_triviality[T: Movable & Copyable]():
     )
 
 
+# ===-----------------------------------------------------------------------===#
+# Implementations tests
+# ===-----------------------------------------------------------------------===#
+
+
+def test_variant_specific_trivialities[trivialities: Int = 0]():
```

Comment:
These tests should live in the according container's test file
- `Varaint` -> `test_variant.mojo`
- `Optional` -> `test_optional.mojo`

Also the tests themselves are quite large. Ideally in these files, each `test_` function should be a unit-style test meaning the scope of them is small. These should get broken down into a couple smaller unit tests if possible 😄